pkg/apis/v1beta1/nodepool.go

```diff
@@ -199,3 +210,79 @@ func (pl *NodePoolList) OrderByWeight() {
 		return ptr.Int32Value(pl.Items[a].Spec.Weight) > ptr.Int32Value(pl.Items[b].Spec.Weight)
 	})
 }
+
+// GetAllowedDisruptions returns the minimum allowed disruptions across all disruption budgets for a given node pool.
+// This returns two values as the resolved value for a percent depends on the number of current node claims.
+func (in *NodePool) GetAllowedDisruptions(ctx context.Context, c clock.Clock) (intstr.IntOrString, intstr.IntOrString, error) {
+	var errs error
+	vals := make([]intstr.IntOrString, len(in.Spec.Disruption.Budgets))
+	for i := range in.Spec.Disruption.Budgets {
+		val, err := in.Spec.Disruption.Budgets[i].GetAllowedDisruptions(c)
+		if err != nil {
+			errs = multierr.Append(errs, err)
+		}
+		vals[i] = val
+	}
+	if errs != nil {
+		return intstr.IntOrString{}, intstr.IntOrString{}, fmt.Errorf("getting nodepool allowed disruptions, %w", errs)
+	}
+	minIntVal, minPercentVal := math.MaxInt64, math.MaxInt64
+	for i := range vals {
+		val := vals[i]
+		// The crontab wasn't active
+		if val.IntVal == -1 {
+			continue
+		}
+		// This returns the percent value if it's a string, and the raw value if it's an int.
+		temp, err := intstr.GetScaledValueFromIntOrPercent(lo.ToPtr(val), 100, false)
+		if err != nil {
+			// Should almost never happen since this is validated when the nodepool is applied
+			return intstr.IntOrString{}, intstr.IntOrString{}, fmt.Errorf("getting intstr scaled value, %w", err)
+		}
+		if val.Type == intstr.Int {
+			minIntVal = lo.Ternary(temp < minIntVal, temp, minIntVal)
+		} else {
+			minPercentVal = lo.Ternary(temp < minPercentVal, temp, minPercentVal)
+		}
+	}
+	// return the values, defaulting to -1 if the value is MaxInt64
+	return intstr.FromInt(lo.Ternary(minIntVal == math.MaxInt64, -1, minIntVal)), intstr.FromString(fmt.Sprintf("%d%%", lo.Ternary(minPercentVal == math.MaxInt64, -1, minPercentVal))), nil
+}
+
+// GetAllowedDisruptions returns an intstr.IntOrString that can be used a comparison
+// for calculating if a disruption action is allowed. It returns an error if the
+// crontab is invalid. This returns -1 if the value is unbounded.
```

Comment:
I wonder if it makes more sense to use MAX_INT here so that what's stored in the value is more indicative of what's actually happening here. Otherwise, we are having to reason about what `-1` means here. Thoughts?